### PR TITLE
Add pipelines

### PR DESCRIPTION
**Pull request by krisvih32**
new pull request since old seems to have broken

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+---
+name: CodeQL
+
+"on":
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: 41 22 * * 3
```
Comment (pkrishnan2) [inline .github/workflows/codeql.yml:21]: Why is there a cron-scehdule for this workflow?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    # https://gh.io/recommended-hardware-resources-for-running-codeql
+    # https://gh.io/supported-runners-and-hardware-resources
+    # https://gh.io/using-larger-runners (GitHub.com only)
+    # Consider using larger runners or machines with greater resources for
+    # possible analysis time improvements.
+    # yamllint disable-lint
+    runs-on:
+      # yamllint disable-line
+      ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```
Comment (pkrishnan2) [inline .github/workflows/codeql.yml:36]: Why matrix is being used

File: .github/workflows/trunk.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        # yamllint disable-line
+        uses: actions/checkout@v4 # Ensure you check out your code
+        with:
+          ref: main
+      - name: Install Trunk
+        run: curl https://get.trunk.io -fsSL | bash
+      - name: temp
+        run: echo "abc "
+      - name: temp 2
+        run: cat .trunk/trunk.yaml
```
Comment (pkrishnan2) [inline .github/workflows/trunk.yaml:28]: Are the above two steps needed?

File: .github/workflows/trunk.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Initialize trunk
+        run: '[ ! -d ".trunk" ] && trunk init ||
+          echo "Trunk is already initialized"'
```
Comment (pkrishnan2) [inline .github/workflows/trunk.yaml:31]: Seems like some of these steps are for debugging, please remove them if not needed

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (krisvih32) [reply .github/workflows/codeql.yml:21]: I realized I don't need it.

Comment (krisvih32) [reply .github/workflows/codeql.yml:21]: Fixed.

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (krisvih32) [reply .github/workflows/codeql.yml:36]: Oh, I am not using matrix or swift.

Comment (krisvih32) [reply .github/workflows/codeql.yml:36]: Fixed.

File: .github/workflows/trunk.yaml (hunk 3 above)
Comment (krisvih32) [reply .github/workflows/trunk.yaml:28]: No, I forgot to remove them from testing.

Comment (krisvih32) [reply .github/workflows/trunk.yaml:28]: Fixed.

File: .github/workflows/trunk.yaml (hunk 4 above)
Comment (krisvih32) [reply .github/workflows/trunk.yaml:31]: No, these are needed.

**Pull request merged**